File: intro-c#/attack/MagicAttack.cs
package attack;

public class MagicAttack : AttackBehaviour {
    private int damage = 4;
    private int loadTime = 5;

    AttackBehaviour() {
        try {
            Thread.sleep(10 * loadTime);
        } catch (InterruptedException e) {
            System.println("Attack was interrupted!");
        }
        System.println(String.format("%s Magic damage dealt!", damage));
    }
}

File: java/1-STRATEGY/attack/MagicAttack.java
package attack;

public class MagicAttack implements IAttackBehaviour {
    private int damage = 4;
    private int loadTime = 5;

    public void attackBehaviour() {
        System.out.println("Starting async magic attack...");
        
        Thread newThread = new Thread(() -> {
            try {
                    Thread.sleep(1000 * loadTime);
                    System.out.println(String.format("%s Magic damage dealt!", damage));
                } catch (InterruptedException e) {
                    System.out.println("Attack was interrupted!");
                }
        });

        newThread.start();
    }
}
